### PR TITLE
✨ [CollectModal] Update UI with auto-follow flow

### PR DESCRIPTION
**Pull request by AuroraHuang22**
截圖 2023-04-12 上午10 50 54
截圖 2023-04-12 上午10 51 05

### CODE REVIEW

File: src/components/EventModal/Collect/index.vue [hunk 1]
```diff
@@ -294,14 +346,35 @@ export default {
         this.fetchUserCollectedCount();
       }
     },
+    uiTxNFTStatus() {
+      if (this.uiTxNFTStatus === 'processing') {
+        if (this.walletHasLoggedIn) {
```
Comment (nwingt) [inline src/components/EventModal/Collect/index.vue:351]: can combine these?

**Review by nwingt — COMMENTED** [review]
I think we need a better name for prompt

File: src/components/EventModal/Collect/index.vue [hunk 2]
```diff
@@ -176,14 +205,20 @@
...
 
 import { logTrackerEvent } from '~/util/EventLogger';
 import { formatNumberWithLIKE } from '~/util/ui';
 
 import clipboardMixin from '~/mixins/clipboard';
 import nftMixin from '~/mixins/nft';
 
+const FOLLOW_STATUS = {
+  FOLLOWED: 'followed', // No need to show any follow UI.
+  UNFOLLOW: 'unFollow', // Show a switch button to toggle follow status.
```
Comment (nwingt) [inline src/components/EventModal/Collect/index.vue:218]:
```suggestion
  UNFOLLOW: 'unfollow', // Show a switch button to toggle follow status.
```

File: src/components/EventModal/Collect/index.vue [hunk 3]
```diff
@@ -176,14 +205,20 @@
...
-import { mapActions } from 'vuex';
+import { mapActions, mapGetters } from 'vuex';
 
 import { logTrackerEvent } from '~/util/EventLogger';
 import { formatNumberWithLIKE } from '~/util/ui';
 
 import clipboardMixin from '~/mixins/clipboard';
 import nftMixin from '~/mixins/nft';
 
+const FOLLOW_STATUS = {
```
Comment (nwingt) [inline src/components/EventModal/Collect/index.vue:216]:
```suggestion
const FOLLOW_PROMPT_STATE = {
```

File: src/components/EventModal/Collect/index.vue [hunk 4]
```diff
@@ -176,14 +205,20 @@
...
+  FOLLOWED: 'followed', // No need to show any follow UI.
+  UNFOLLOW: 'unFollow', // Show a switch button to toggle follow status.
+  PROMPT: 'prompt', // Show auto-followed UI.
```
Comment (nwingt) [inline src/components/EventModal/Collect/index.vue:219]:
```suggestion
  AUTO: 'auto', // Show auto-followed UI.
```

File: src/components/EventModal/Collect/index.vue [hunk 5]
```diff
@@ -226,9 +261,17 @@ export default {
       justCollectedNFTId: undefined,
       shouldShowMessageInput: false,
       memo: '',
+      followStatus: FOLLOW_STATUS.UNFOLLOW,
```
Comment (nwingt) [inline src/components/EventModal/Collect/index.vue:264]:
```suggestion
      followPromptState: FOLLOW_PROMPT_STATE.UNFOLLOW,
```

File: src/components/EventModal/Collect/index.vue [hunk 6]
```diff
@@ -226,9 +261,17 @@ export default {
...
+      isFollowStatusChanging: false,
```
Comment (nwingt) [inline src/components/EventModal/Collect/index.vue:265]:
```suggestion
      isFollowPromptUpdating: false,
```

File: src/components/EventModal/Collect/index.vue [hunk 7]
```diff
@@ -273,6 +316,15 @@ export default {
         this.getUserInfoByAddress(this.iscnOwner)?.displayName || 'creator'
       );
     },
+    followButtonText() {
```
Comment (nwingt) [inline src/components/EventModal/Collect/index.vue:319]:
```suggestion
    followPromptButtonText() {
```

File: src/store/modules/wallet.js [hunk 8]
```diff
@@ -170,6 +177,7 @@ const getters = {
   getLikerInfo: state => state.likerInfo,
   walletFollowees: state => state.followees,
   walletFollowers: state => state.followers,
+  walletPromptedList: state => state.interactedCreators,
```
Comment (nwingt) [inline src/store/modules/wallet.js:180]:
```suggestion
  walletInteractedCreators: state => state.interactedCreators,
```

File: src/store/modules/wallet.js [hunk 9]
```diff
@@ -564,6 +572,7 @@ const actions = {
       commit(WALLET_SET_USER_INFO, null);
       commit(WALLET_SET_FOLLOWEES, []);
       commit(WALLET_SET_FOLLOWERS, []);
+      commit(WALLET_SET_PROMPTED_LIST, []);
```
Comment (nwingt) [inline src/store/modules/wallet.js:575]:
```suggestion
      commit(WALLET_SET_INTERACTED_CREATORS, []);
```

File: src/store/modules/wallet.js [hunk 10]
```diff
@@ -642,6 +659,16 @@ const actions = {
       throw error;
     }
   },
+  async walletAddPromptCreator({ state, commit }, creator) {
```
Comment (nwingt) [inline src/store/modules/wallet.js:662]:
```suggestion
  async walletAddInteractedCreator({ state, commit }, creator) {
```

File: src/store/modules/wallet.js [hunk 11]
```diff
@@ -642,6 +659,16 @@ const actions = {
...
+    const prevPromptedList = state.interactedCreators;
```
Comment (nwingt) [inline src/store/modules/wallet.js:663]:
```suggestion
    const prevInteractedCreators = state.interactedCreators;
```

File: src/server/api/routes/users/v2/follow.js [hunk 12]
```diff
@@ -43,45 +41,67 @@ router.get('/followees', authenticateV2Login, async (req, res, next) => {
...
-        logType: 'UserCreatorFollow',
-        type: 'wallet',
-        user,
-        creatorWallet: creator,
+router.get('/interacted', authenticateV2Login, async (req, res, next) => {
+  try {
+    const { user } = req.session;
+    const userDoc = await walletUserCollection.doc(user).get();
+    if (!userDoc.exists) {
+      res.json({ followees: [] });
```
Comment (nwingt) [inline src/server/api/routes/users/v2/follow.js:49]: Should be same as below?
```suggestion
      res.json({ interactedCreators: [] });
```

File: src/store/modules/wallet.js [hunk 13]
```diff
@@ -605,8 +614,16 @@ const actions = {
...
-        dispatch('lazyGetUserInfoByAddresses', followees);
+      const { interactedCreators } = await this.$axios.$get(
+        getUserV2InteractedCreators()
+      );
+      commit(WALLET_SET_INTERACTED_CREATORS, interactedCreators);
+
+      if (followees.length || interactedCreators.length) {
+        dispatch('lazyGetUserInfoByAddresses', [
+          ...followees,
+          ...interactedCreators,
```
Comment (nwingt) [inline src/store/modules/wallet.js:625]: should make sure these are unique?

**Review by nwingt — COMMENTED** [review]
Please resolve conflicts too

File: src/components/EventModal/Collect/index.vue [hunk 14]
```diff
@@ -37,16 +37,69 @@
...
+          :path="
+            hasConnectedWallet
+              ? 'tx_modal_status_complete_text_collect'
+              : 'tx_modal_status_complete_text_collect_without_wallet'"
+        >
+          <NuxtLink
+            class="font-[600] underline"
+            place="portfolio"
+            :to="localeLocation({
+              name: 'id___en',
```
Comment (nwingt) [inline src/components/EventModal/Collect/index.vue:51]: Why adding language suffix?
```suggestion
              name: 'id___en',
```

File: src/server/api/routes/users/v2/follow.js [hunk 15]
```diff
@@ -114,6 +106,7 @@ router.delete('/followees', authenticateV2Login, async (req, res, next) => {
       creatorWallet: creator,
     });
     res.sendStatus(200);
+    res.sendStatus(200);
```
Comment (williamchong) [inline src/server/api/routes/users/v2/follow.js:109]: duplicated
```suggestion
```

File: src/server/api/routes/users/v2/follow.js [hunk 16]
```diff
@@ -23,7 +20,11 @@ router.get('/followees', authenticateV2Login, async (req, res, next) => {
       res.json({ followees: [] });
       return;
     }
-    const { followees: walletFollowees = [], email } = userDoc.data();
+    const {
+      followees: walletFollowees = [],
+      email,
+      interactedCreators = [],
```
Comment (williamchong) [inline src/server/api/routes/users/v2/follow.js:26]: How about call them `followees` and `pastFollowees` ?

File: src/store/modules/wallet.js [hunk 17]
```diff
@@ -642,16 +655,35 @@ const actions = {
...
+  //     await this.$api.$post(postUserV2InteractedCreators(creator));
+  //     commit(WALLET_SET_INTERACTED_CREATORS, [
+  //       ...state.interactedCreators,
+  //       creator,
+  //     ]);
+  //   } catch (error) {
+  //     commit(WALLET_SET_INTERACTED_CREATORS, prevInteractedCreators);
+  //     throw error;
+  //   }
+  // },
```
Comment (williamchong) [inline src/store/modules/wallet.js:670]: Please remove if not used

File: src/store/modules/wallet.js [hunk 18]
```diff
@@ -642,16 +655,35 @@ const actions = {
...
+    const prevInteractedCreators = state.interactedCreators;
     try {
       await this.$api.$delete(deleteUserV2Followees(creator));
       commit(
         WALLET_SET_FOLLOWEES,
         [...state.followees].filter(followee => followee !== creator)
       );
+      commit(WALLET_SET_INTERACTED_CREATORS, [
+        ...state.interactedCreators,
+        creator,
```
Comment (williamchong) [inline src/store/modules/wallet.js:682]: Would `state.interactedCreators.concat(creator)` be simpler?

File: src/components/EventModal/Collect/index.vue [hunk 19]
```diff
@@ -296,14 +402,35 @@ export default {
         this.fetchUserCollectedCount();
       }
     },
+    uiTxNFTStatus() {
+      if (this.uiTxNFTStatus === 'processing' && this.walletHasLoggedIn) {
+        const creator = this.getNFTClassMetadataById(this.classId)?.iscn_owner;
+        if (
+          this.walletFollowees?.includes(creator) ||
+          creator === this.getAddress
```
Comment (williamchong) [inline src/components/EventModal/Collect/index.vue:410]: Technically `creator === this.getAddress` is not a `followed` relationship, you don't get an email per your own nft mint. Not sure if we should just hide the whole UI here.
but this is design's decision to keep this or not

**Review by williamchong — CHANGES_REQUESTED** [review]

File: src/components/EventModal/Collect/index.vue (hunk 19 above)
Comment (AuroraHuang22) [reply src/components/EventModal/Collect/index.vue:410]: Kiu said that when collecting your own work, don't display the "follow" related UI.

Comment (williamchong) [reply src/components/EventModal/Collect/index.vue:410]: So state.followed means hidden in this component?

Comment (AuroraHuang22) [reply src/components/EventModal/Collect/index.vue:410]: If state.followed, then display the default UI.. 
Maybe I should change state.followed to state.default.
截圖 2023-04-17 下午6 29 43

**Review by nwingt — APPROVED** [review]

**Comment by AuroraHuang22** [comment]
截圖 2023-04-19 下午12 24 07
@nwingt

**Review by nwingt — APPROVED** [review]

**Pull request merged**